feat(sol): keccak proofs are not padded

Maddiaa0 opened the pull request:
Verifier optimization work

2^20 circuit goes from 2120372 -> 1908766 so 211k savings on the non optimized verifier

Maddiaa0 committed:
chore: reduce keccak proofs to log_n not const_proof_size_log_n

Maddiaa0 commented on barretenberg/cpp/src/barretenberg/dsl/acir_proofs/honk_contract.hpp:
```diff
@@ -1727,9 +1728,9 @@ abstract contract BaseHonkVerifier is IVerifier {
-        if (proof.length != PROOF_SIZE * 32) {
-            revert ProofLengthWrong();
-        }
+        // if (proof.length != PROOF_SIZE * 32) {
```
todo

zac-williamson replied:
is this todo going to be completed before the PR is merged? if not could you create an issue and document in a comment what needs to be done?

ledwards2225 replied:
this has been reinstated

Maddiaa0 committed:
chore: fix wasm build

Maddiaa0 committed:
chore: fix failing ecMul

lucasxia01 commented on barretenberg/cpp/src/barretenberg/sumcheck/sumcheck.hpp:
```diff
@@ -702,16 +713,32 @@ template <typename Flavor, size_t virtual_log_n = CONST_PROOF_SIZE_LOG_N> class
+        if constexpr (!IsAnyOf<Flavor, UltraKeccakFlavor, UltraKeccakZKFlavor>) {
+            log_n = virtual_log_n;
+        } else {
+            // Set log_n to the highest non zero value in padding_indicator_array
```
hm this is not great, ideally we could just add a size_t to the padding indicator array or set this value in the sumcheck class

ledwards2225 replied:
handled

lucasxia01 commented on barretenberg/sol/src/honk/BaseHonkVerifier.sol:
```diff
@@ -53,20 +52,20 @@ abstract contract BaseHonkVerifier is IVerifier {
-        if (proof.length != PROOF_SIZE * 32) {
-            revert ProofLengthWrong();
-        }
+        // if (proof.length != PROOF_SIZE * 32) {
```
need a todo here

ledwards2225 replied:
check is reinstated

Maddiaa0 committed:
chore: use is padding for everything

Commit:
[empty] Start merge-train. Choo choo.

Commit:
Merge branch 'next' into merge-train/barretenberg

Commit:
Merge branch 'next' into merge-train/barretenberg

Commit:
Merge branch 'next' into merge-train/barretenberg

ledwards2225 committed:
format revert

ledwards2225 committed:
fmt

Commit:
Merge branch 'next' into merge-train/barretenberg

Commit:
Merge branch 'next' into merge-train/barretenberg

Commit:
Merge branch 'next' into merge-train/barretenberg

Commit:
Merge branch 'next' into merge-train/barretenberg

Commit:
Merge branch 'next' into merge-train/barretenberg

Commit:
Merge branch 'next' into merge-train/barretenberg

Commit:
feat: remove duplication of hiding circuit creation by handling its completion outside of ClientIVC::prove (#16135)

This PR changes the way the construction of the hiding kernel is handled
to accommodate for the hiding kernel being implemented in Noir.
Changes include: 
- added two methods `construct_and_accumulate_next_circuit` and
`construct_hiding_circuit` to
`PrivateFunctionExecutionMockCircuitProducer`, to handle creation of the
hiding circuit in our testing modules
- removed the duplication of `complete_hiding_circuit_logic`. Now the
`hiding_circuit` builder is saved as a member `ClientIVC` class.
- added a `construct_and_accumulate_mock_kernel` method to
`ivc_recursion_constraints` test module.
- updated the API according to the changes to the `ClientIVC` class.

---------
Co-authored-by: Khashayar Barooti <khashayar@aztecprotocol.com>

Commit:
chore: remove is_kernel from MegaCircuitBuilder (#16186)

Remove `is_kernel` flag from the MegaCircuitBuilder as the functionality
it provides (knowing when we are building a kernel circuit) can be
handled solely in the ClientIVC class.

Commit:
Merge branch 'next' into merge-train/barretenberg

Commit:
Merge branch 'next' into merge-train/barretenberg

Commit:
Merge branch 'next' into merge-train/barretenberg

federicobarbacovi committed:
fix: Fourth wire in ram/rom is not accumulated (#16142)

The current version of the `add_ram_rom_memory_records_to_wire_4`
function accumulates values on the fourth wire (`wires[3] += ...`). This
means that running the `OinkProver` twice on a proving key with ram/rom
gates will produce two different outputs. This PR fixes this bug. Closes
https://github.com/AztecProtocol/barretenberg/issues/1497

---------

Co-authored-by: AztecBot <tech@aztecprotocol.com>

lucasxia01 committed:
feat!: Remove unnecessary gates when setting default pairing points as public inputs (#14448)

This used to curiously fail in the IVC integration tests, but no longer
does...

This PR removes the gates that were unnecessarily created for the
default pairing points from the bigfield constructor. Instead, we create
the constant witnesses directly. This reduces the amount of gates from
~3500 to 20.

Closes https://github.com/AztecProtocol/barretenberg/issues/984.

ludamad committed:
Revert "chore: default turn on vk cache (#16209)"

This reverts commit 836cfefcb8de5f99ebdcd80a1b0a7afeb728dc88.

ledwards2225 committed:
chore: change how virtual_log_n is set in sumcheck (#16215)

Set `virtual_log_n` in sumcheck based on an input parameter rather than
a template parameter. This allows setting the value at runtime (i.e.
when the value is not known at compile time) which is useful for
removing the padding rounds from the solidity verifier.

ledwards2225 committed:
Merge branch 'merge-train/barretenberg' into md/sol-reduce-transcript-hashing

ledwards2225 committed:
Merge branch 'merge-train/barretenberg' into md/sol-reduce-transcript-hashing

ledwards2225 committed:
format sol + cpp

ledwards2225 committed:
all tests passing (expect for proof size tests)

ledwards2225 committed:
proof size as a function of log_n

ledwards2225 committed:
fix use of bytesToG1ProofPoint in sol

ledwards2225 committed:
update cpp

Commit:
[empty] Start merge-train. Choo choo.

Commit:
Merge branch 'next' into merge-train/barretenberg

ledwards2225 committed:
update ULTRA_KECCAK_PROOF_LENGTH value

ledwards2225 committed:
comment

ledwards2225 committed:
Merge branch 'merge-train/barretenberg' into md/sol-reduce-transcript-hashing

ledwards2225 committed:
update more constants; probably need flexible formula

ledwards2225 committed:
some debug code for proof size

Maddiaa0 committed:
fix: dirty memory sorry

Maddiaa0 committed:
chore: remove srings lib

Maddiaa0 committed:
update honk contract

ledwards2225 committed:
Merge branch 'merge-train/barretenberg' into md/sol-reduce-transcript-hashing

ledwards2225 committed:
temporarily disable proof size checks in ts

ledwards2225 committed:
update proof size checking logic in sol, remove from bbjs test

ledwards2225 commented on barretenberg/acir_tests/bbjs-test/src/index.ts:
```diff
@@ -8,13 +8,6 @@ const logger = pino({
 });
 
-const UH_PROOF_FIELDS_LENGTH = 508; // length of UltraZKHonk flavor
-const UH_KECCAK_PROOF_ELEMENTS_LENGTH = 426; // length of UltraZKHonk flavor
```
Unfortunately this proof size is now a function of the log dyadic circuit size and it seems too complicated to implement that here. Even if we could, it's not clear it's worth the complex code duplication.

ledwards2225 committed:
clean

ledwards2225 reviewed (approved):
I've reviewed this indirectly through spending a chunk of time updating the bberg. LG, lets get it in
